### ISSUE TITLE
New Rule: no bangbang (!!) operators

### ISSUE THREAD

**Issue by almercier** [eslint/eslint#15823]
### Rule details

Disallow the use of "!!value" for converting a value to boolean, preferring `Boolean(value)` instead.

### Related ECMAScript feature

13.5.7, 20.3.1.1

### What type of rule is this?

Enforces a formatting/stylistic preference

### Example code

```js
!!foo
!!!!!!!!foo
```

### Why should this rule be in the core instead of a plugin?

There are several discussions online on whether people prefer using !! or Boolean() (or allowing both) to cast values to booleans, but our team believes that Boolean is much more explicit and readable. I'd like to be able to enforce this in our eslint settings to ensure consistency in our code.

Ideally this rule would also enforce  `!!!val` or `!!!!!!!!val` which I really hope no-one would ever do...

By default should be disabled, but have the ability to enforce only !! or only Boolean().

### Participation

- [ ] I am willing to submit a pull request to implement this rule.

**Comment by mdjermanovic** [comment]
Hi @almercier, thanks for the issue!

> Disallow the use of "!!value" for converting a value to boolean, preferring `Boolean(value)` instead.

This rule already exists: [no-implicit-coercion](https://eslint.org/docs/rules/no-implicit-coercion)

Here's an [online demo example](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLWltcGxpY2l0LWNvZXJjaW9uOiBcImVycm9yXCIgKi9cblxuISF2YWx1ZSIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6ImxhdGVzdCIsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnt9fX0=) that demonstrates this functionality.

**Comment by almercier** [comment]
Aha,  sorry about that! That rule does exactly what I need, just couldn't find it in the site search.
